app/controllers/Controller.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import javax.inject.{Inject, Singleton}
+import model.{PeriodKey, Vrn, VrtId, VrtRepaymentDetailData}
+import play.api.Logger
+import play.api.libs.json.Json
+import play.api.mvc.{Action, AnyContent, ControllerComponents}
+import repository.VrtRepo
+import uk.gov.hmrc.play.bootstrap.controller.BackendController
+import reactivemongo.bson.BSONObjectID
+
+import scala.concurrent.ExecutionContext
+
+@Singleton
+class Controller @Inject() (cc: ControllerComponents, vrtRepo: VrtRepo)(implicit executionContext: ExecutionContext) extends BackendController(cc) {
+
+  def storeRepaymentData(): Action[VrtRepaymentDetailData] = Action.async(parse.json[VrtRepaymentDetailData]) { implicit request =>
+    Logger.debug(s"received ${request.body.toString}")
+    for {
+      data <- vrtRepo.findByVrnAndPeriodKeyAndRiskingStatus(request.body.vrn,
+                                                            PeriodKey(request.body.repaymentDetailsData.periodKey),
+                                                            request.body.repaymentDetailsData.riskingStatus)
+      vrtId: VrtId = if (data.size == 0) VrtId.fresh else data(0)._id.getOrElse(throw new RuntimeException("No id"))
```

Comment:
better:
 new RuntimeException("No id, body:[${request.body}]")
so if exception happens we will know more about the context